packages/client/hmi-client/src/petrinet/petrinet-renderer.ts

```diff
@@ -276,6 +277,8 @@ export class PetrinetRenderer extends BasicRenderer<NodeData, EdgeData> {
 
 		// (Re)create dragging listeners
 		this.on('node-drag-start', (_eventName, event, selection: D3SelectionINode<NodeData>) => {
+			selection.select('circle').attr('stroke', highlightedStrokeColour);
+			selection.select('rect').attr('stroke', highlightedStrokeColour);
```

Comment:
This almost works but not quite right, you cannot arbitrarily select circle/rect as I never give any guarantee which circle/rect you are going to get. e.g. you are changing the style of things you don't want. Note below the right anchor is larger.

<img width="144" alt="image" src="https://user-images.githubusercontent.com/1220927/229931971-07f754bb-4ce2-46d4-a4c6-1571b67e26c0.png">


So if we take a transition-node for example, it is made up of
- 1 rect
- 4 circles (the handles)
- text

Your selection logic would change the rect and the circle, in this case the first circle which happens to be the east-facing one.

So you probably want to add a specific class selector such that the shape can be uniquely identified.

Reply:
Good catch, thanks!